app/lib/submission_builder/financial_transaction.rb

```diff
@@ -26,8 +26,19 @@ def document
             xml.RoutingTransitNumber sanitize_for_xml(@submission.data_source.routing_number) if @submission.data_source.routing_number.present?
             xml.BankAccountNumber sanitize_for_xml(@submission.data_source.account_number) if @submission.data_source.account_number.present?
             xml.PaymentAmount @submission.data_source.withdraw_amount if @submission.data_source.withdraw_amount.present?
+            xml.AccountHolderType "2" if @submission.data_source.requires_additional_debit_information?
             xml.RequestedPaymentDate date_type(@submission.data_source.date_electronic_withdrawal) if @submission.data_source.date_electronic_withdrawal.present?
             xml.NotIATTransaction 'X'
+            if @submission.data_source.requires_additional_debit_information?
+              xml.AddendaRecord do
+                xml.TaxTypeCode "01000"
+                xml.TaxPeriodEndDate "2024-12-31"
```

Comment:
do we need to use `date_type(Date.new(@submission.data_source.tax_return_year, 12, 31))` for the year to be the tax return year each time?

Reply:
ooh good call